cmd/pitr/collector/collector.go

```diff
@@ -164,14 +165,29 @@ func (c *Collector) CollectBinLogs() error {
 	return nil
 }
 
-// minio.PutObject method would check if data implements `os.File` and handle it respectively.
-// So we have to restrain it to `io.Reader` only.
-type reader struct {
-	r io.Reader
+type pipeReader struct {
+	f        *os.File
+	buf      *bytes.Buffer
+	notEmpty bool
 }
 
-func (r *reader) Read(p []byte) (int, error) {
-	return r.r.Read(p)
+func (p *pipeReader) ReadToBuf() {
+	b := make([]byte, 1024)
+	for {
+		n, err := p.f.Read(b)
+		if err == io.EOF {
+			if !p.notEmpty {
+				time.Sleep(10 * time.Microsecond)
+				continue
+			}
+			break
+		}
+		if n == 0 {
+			continue
+		}
+		p.buf.Write(b[:n])
```

Comment:
should we reset `b`?

Reply:
no need, we only read bytes that were [over]written